src/main/scala/com/johnsnowlabs/ml/openvino/OpenvinoWrapper.scala

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2017-2022 John Snow Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.johnsnowlabs.ml.openvino
+
+import com.johnsnowlabs.ml.util.LoadExternalModel.notSupportedEngineError
+import com.johnsnowlabs.ml.util.{ONNX, Openvino, TensorFlow}
+import com.johnsnowlabs.util.{ConfigHelper, ConfigLoader, FileHelper, ZipArchiveUtil}
+import org.apache.commons.io.{FileUtils, FilenameUtils}
+import org.apache.spark.SparkFiles
+import org.apache.spark.sql.SparkSession
+import org.intel.openvino.Openvino.save_model
+import org.intel.openvino.{CompiledModel, Core, Model}
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.io.File
+import java.nio.file.{Files, Path, Paths}
+import java.util.UUID
+import scala.collection.JavaConverters._
+
+class OpenvinoWrapper(var modelName: Option[String] = None) extends Serializable {
+
+  /** For Deserialization */
+  def this() = {
+    this(null)
+  }
+
+  // Important for serialization on none-kyro serializers
+  @transient private var compiledModel: CompiledModel = _
+
+  def getCompiledModel(
+      properties: Map[String, String] = Map.empty[String, String]): CompiledModel =
+    this.synchronized {
+      if (compiledModel == null) {
+        val modelPath = SparkFiles.get(s"${modelName.get}.xml")
+        val device = ConfigLoader.getConfigStringValue(ConfigHelper.openvinoDevice)
```

Comment:
This line is raising the following exception in cluster environments:
```org.apache.spark.SparkException: Job aborted due to stage failure: Task 7 in stage 0.0 failed 4 times, most recent failure: Lost task 7.3 in stage 0.0 (TID 10) (10.222.88.147 executor 1): java.lang.IllegalStateException: SparkSession should only be created and accessed on the driver.```

I suggest sending CPU by default and letting personalization for a future release

Reply:
Hi @danilojsl 
Let's do that. Please adjust this part and leave a TODO for later

Reply:
Reverted this and defaulting to CPU. Thanks!